### PR TITLE
DOCSP-45715: Sealed class v3.0 changes

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-45715> and <https://jira.mongodb.org/browse/DOCSP-45716>

### Staging Links

<li><a href=https://deploy-preview-345--mongodb-docs-csharp.netlify.app/fundamentals/serialization>fundamentals/serialization</a></li><li><a href=https://deploy-preview-345--mongodb-docs-csharp.netlify.app/upgrade/v3>upgrade/v3</a></li>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/upgrade/v3.txt [hunk 1]
```diff
@@ -203,4 +203,11 @@ Version 3.0 Breaking Changes
...
 - The driver removes individual cluster events from ``MongoClient.Cluster``. To listen for
-  cluster events, use `ClusterBuilder.Subscribe() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.Core.Configuration.ClusterBuilder.Subscribe.html>`__.
\ No newline at end of file
+  cluster events, use `ClusterBuilder.Subscribe() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.Core.Configuration.ClusterBuilder.Subscribe.html>`__.
+
+- All concrete serializers have been sealed. To implement a custom serializer, implement the
+  ``IBsonSerializer`` interface.
+
+- The ``MongoClient``, ``MongoDatabase``, and ``MongoCollection`` classes have been
+  sealed. We recommend using the ``IMongoClient``, ``IMongoDatabase``, and
```
Comment (norareidy) [inline source/upgrade/v3.txt:212]: S: I'd update to active voice

```suggestion
- The driver seals all concrete serializers. To implement a custom serializer, implement the
  ``IBsonSerializer`` interface.

- The driver seals the ``MongoClient``, ``MongoDatabase``, and ``MongoCollection`` classes. We recommend using the ``IMongoClient``, ``IMongoDatabase``, and
```

**Review by norareidy — APPROVED** [review]
LGTM + one suggestion!

File: source/upgrade/v3.txt [hunk 2]
```diff
@@ -203,4 +203,11 @@ Version 3.0 Breaking Changes
     property instead.
 
 - The driver removes individual cluster events from ``MongoClient.Cluster``. To listen for
-  cluster events, use `ClusterBuilder.Subscribe() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.Core.Configuration.ClusterBuilder.Subscribe.html>`__.
\ No newline at end of file
+  cluster events, use `ClusterBuilder.Subscribe() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.Core.Configuration.ClusterBuilder.Subscribe.html>`__.
+
+- The driver seals all concrete serializers. To implement a custom serializer, implement the
```
Comment (BorisDog) [inline source/upgrade/v3.txt:208]: We had some questions of "why" classes got sealed.
It's worth adding a general comment followed by these concrete examples:
Something like "Some of the types that were not designed for extension via inheritance became sealed. Starting with 3.0 we recommend using "composition over inheritance" approach for extending the driver functionality."

**Review by BorisDog — CHANGES_REQUESTED** [review]

**Review by BorisDog — APPROVED** [review]
LGTM

**Pull request merged**